SetDbSessionVars listener crashes when database is not connected

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Version

11.0.5-bugfixes

### Bug description

## Steps to reproduce

1. Pull the latest `11.0/bugfixes` branch (after PR #22281)
2. Run `make test-db-install` or `make phpunit`

## Root cause

`SetDbSessionVars::onPostBoot()` does not check if the database connection is established before calling `$DB->setTimezone()`.

Introduced by #22281





### Relevant log output

```shell
Error {#107
    #message: "mysqli object is not fully initialized"
    #code: 0
    #file: "/var/www/glpi/src/DBmysql.php"
    #line: 356
    trace: {
      /var/www/glpi/src/DBmysql.php:356 {
        DBmysql->escape($string)^
        › {
        ›     return $this->dbh->real_escape_string((string) $string);
        › }
        arguments: {
          $string: "UTC"
        }
      }
      /var/www/glpi/src/DBmysql.php:1947 { …}
      /var/www/glpi/src/DBmysql.php:1872 { …}
      /var/www/glpi/src/Glpi/Kernel/Listener/PostBootListener/SetDbSessionVars.php:61 { …}
      /var/www/glpi/vendor/symfony/event-dispatcher/Debug/WrappedListener.php:116 { …}
      /var/www/glpi/vendor/symfony/event-dispatcher/EventDispatcher.php:220 { …}
      /var/www/glpi/vendor/symfony/event-dispatcher/EventDispatcher.php:56 { …}
      /var/www/glpi/vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php:139 { …}
      /var/www/glpi/src/Glpi/Kernel/Kernel.php:149 { …}
      /var/www/glpi/src/Glpi/Console/Application.php:121 { …}
      /var/www/glpi/bin/console:143 { …}
    }
  }
  make: *** [Makefile:138 : test-db-install] Erreur 255
```

### Page URL

_No response_

### Steps To reproduce

_No response_

### Your GLPI setup information

_No response_

### Anything else?

_No response_

--- a/src/Glpi/Kernel/Listener/PostBootListener/SetDbSessionVars.php
+++ b/src/Glpi/Kernel/Listener/PostBootListener/SetDbSessionVars.php
@@ -34,6 +34,7 @@
 
 namespace Glpi\Kernel\Listener\PostBootListener;
 
+use DBConnection;
 use Glpi\Debug\Profiler;
 use Glpi\Kernel\ListenersPriority;
 use Glpi\Kernel\PostBootEvent;
@@ -56,7 +57,7 @@ final class SetDbSessionVars implements EventSubscriberInterface
 
         Profiler::getInstance()->start('SetDbSessionVars::execute', Profiler::CATEGORY_BOOT);
 
-        if ($DB->use_timezones) {
+        if (DBConnection::isDbAvailable() && $DB->use_timezones) {
             $timezone = $this->getConfiguredTimezone();
             $DB->setTimezone($timezone);
         }

--- a/tests/functional/Glpi/Kernel/Listener/PostBootListener/SetDbSessionVarsTest.php
+++ b/tests/functional/Glpi/Kernel/Listener/PostBootListener/SetDbSessionVarsTest.php
@@ -97,4 +97,24 @@ class SetDbSessionVarsTest extends GLPITestCase
 
         $this->assertEquals($expected_timezone, $result['tz']);
     }
+
+    public function testOnPostBootWithDisconnectedDatabase(): void
+    {
+        global $DB;
+
+        $db_backup = $DB;
+
+        try {
+            $DB = null;
+
+            $_SESSION['glpitimezone'] = 'Europe/Paris';
+
+            $listener = new SetDbSessionVars();
+            $listener->onPostBoot();
+
+            $this->assertTrue(true);
+        } finally {
+            $DB = $db_backup;
+        }
+    }
 }

